### PR TITLE
Add Golden master test

### PR DESCRIPTION
**Pull request by bhugot**
As we don't have full test for instrumentation, I have created a golden master test so we can refactor instrumentation

### CODE REVIEW

**Comment by bhugot** [comment]
@lucaslorentz please merge this so that I could work on refactoring instrumentation to be more testable. And I could upgrade this test on my other PR.

**Comment by lucaslorentz** [comment]
@bhugot 
There more than 120 files on this PR. 
Why do we need the other ApprovalTest library code?

**Comment by lucaslorentz** [comment]
@bhugot 
There more than 120 files on this PR. 
Why do we need code from library ApprovalTest library instead of NugetPackage?

**Comment by bhugot** [comment]
because the package is not netstandard so no build on the ci. I m working on this too

**Comment by lucaslorentz** [comment]
Can you create a test project targeting full framework (NET46 or NET47)?
And don't include that test project on CI until ApprovalTest support netstandard?

**Comment by bhugot** [comment]
I could but if not playing it on CI no reason to have it

**Comment by bhugot** [comment]
BTW I just did the PR on ApprovalTest.

**Comment by lucaslorentz** [comment]
What do you think about those options?
- Setup windows CI to run ApprovalTests on NET 4.7.1
- Replace ApprovalTests by a minimalist implementation with only what you need, 2 text files comparison.

I created a PR to help you with option 1: https://github.com/bhugot/minicover/pull/5
Build output: https://ci.appveyor.com/project/lucaslorentz/minicover/build/4

It's nice to run our builds on windows also, to specially the sample project.

**Comment by bhugot** [comment]
Better than nothing ;)

File: tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace ApprovalTests.Reporters
+{
+    public class TravisCIReporter : IEnvironmentAwareReporter
+    {
+        public static readonly TravisCIReporter INSTANCE = new TravisCIReporter();
+
+        public void Report(string approved, string received)
+        {
+            ContinousDeliveryUtils.ReportOnServer(approved,received);
```
Comment (lucaslorentz) [inline tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs]: I'm curious to understand what this line does.
Does it show results in any special way on TravisCI?

**Comment by lucaslorentz** [comment]
I'm ok with merging this.
Do you want to adjust something after my changes?

File: tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+using System;
+using ApprovalTests.Core;
+
+namespace ApprovalTests.Reporters
+{
+    public class TravisCIReporter : IEnvironmentAwareReporter
```
Comment (lucaslorentz) [inline tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs]: This file was inside _ApprovalTests folder, but it doesn't exist on ApprovalTests library.

File: tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs (hunk 1 above)
Comment (bhugot) [reply tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs]: It should for me but not sure it's working

File: tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs (hunk 2 above)
Comment (bhugot) [reply tests/MiniCover.ApprovalTests/_ApprovalTests/Reporters/TravisCIReporter.cs]: Yeah I m going to remove it

**Comment by lucaslorentz** [comment]
Btw, do you develop on windows or linux?
I'm wondering if you will be able to run the ApprovalTests during development or will have to wait CI.

**Comment by bhugot** [comment]
I m using windows 10 ubuntu bash to see how it's working on linux so I can execute on both.
But as it is, it should not matter as we do the transformation to linux way

**Comment by lucaslorentz** [comment]
I'm going to squash commits during merge, to avoid a lot of files added and deleted on the history.

**Comment by bhugot** [comment]
yep no problem ;p I still have my branch if need to look at something

**Pull request merged**